### PR TITLE
Creates Clean branch

### PR DESCRIPTION
**Pull request by JoseAlcerreca**

### CODE REVIEW

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskViewModel.kt [hunk 1]
```diff
@@ -37,7 +38,8 @@ import kotlinx.coroutines.launch
  * how to deal with more complex scenarios.
  */
 class AddEditTaskViewModel(
-    private val tasksRepository: TasksRepository
+    private val getTaskUseCase: GetTaskUseCase,
```
Comment (manuelvicnt) [inline app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskViewModel.kt:41]: To make it simpler and read better, you can call these `getTask` and `save` without the `UseCase` suffix

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/DefaultTasksRepository.kt [hunk 2]
```diff
@@ -45,34 +43,31 @@ class DefaultTasksRepository(
...
+                }
+            }
 
-               val newTasks = fetchTasksFromRemoteOrLocal(forceUpdate)
+            val newTasks = fetchTasksFromRemoteOrLocal(forceUpdate)
 
-               // Refresh the cache with the new tasks
-               (newTasks as? Success)?.let { refreshCache(it.data) }
+            // Refresh the cache with the new tasks
+            (newTasks as? Success)?.let { refreshCache(it.data) }
```
Comment (manuelvicnt) [inline app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/DefaultTasksRepository.kt:57]: I think this logic can be simpler. It's hard to follow when it's been a success and when it hasn't been. It took me a bit of time to go through the code and realise that from line 57 to 67 is the happy path (even having to cast it to Success twice) and only the last line covers the sad path. I think it'd be more understandable if in line 57 there's an if-else statement depending on the Result state.

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailViewModel.kt [hunk 3]
```diff
@@ -35,7 +38,11 @@ import kotlinx.coroutines.launch
  * Fragment's actions listener.
  */
 class TaskDetailViewModel(
-    private val tasksRepository: TasksRepository
+    private val getTaskUseCase: GetTaskUseCase,
```
Comment (manuelvicnt) [inline app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailViewModel.kt:41]: Ditto

**Review by manuelvicnt — APPROVED** [review]
Minor comments. LGTM

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskViewModel.kt (hunk 1 above)
Comment (JoseAlcerreca) [reply app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskViewModel.kt:41]: I think for an example this is more clear, but don't have strong feelings about it.

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/DefaultTasksRepository.kt (hunk 2 above)
Comment (JoseAlcerreca) [reply app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/DefaultTasksRepository.kt:57]: Comes from the master branch, so let's keep it the same for now and we can fix master later.
